server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationClusterSettingIT.java

```diff
@@ -113,17 +115,97 @@ public void testReplicationWithSegmentReplicationClusterSetting() throws Excepti
         }
     }
 
-    public void testIndexReplicationSettingOverridesClusterSetting() throws Exception {
+    public void testIndexReplicationSettingOverridesSegRepClusterSetting() throws Exception {
+        Settings settings = Settings.builder().put(CLUSTER_SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT).build();
+        final String ANOTHER_INDEX = "test-index";
+
         // Starting two nodes with primary and replica shards respectively.
-        final String primaryNode = internalCluster().startNode();
+        final String primaryNode = internalCluster().startNode(settings);
         prepareCreate(
             INDEX_NAME,
             Settings.builder()
                 // we want to override cluster replication setting by passing a index replication setting
                 .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.DOCUMENT)
         ).get();
-        ensureYellowAndNoInitializingShards(INDEX_NAME);
+        createIndex(ANOTHER_INDEX);
+        ensureYellowAndNoInitializingShards(INDEX_NAME, ANOTHER_INDEX);
+        final String replicaNode = internalCluster().startNode(settings);
+        ensureGreen(INDEX_NAME, ANOTHER_INDEX);
+
+        final int initialDocCount = scaledRandomIntBetween(20, 30);
+        for (int i = 0; i < initialDocCount; i++) {
+            client().prepareIndex(INDEX_NAME).setId(Integer.toString(i)).setSource("field", "value" + i).execute().actionGet();
+            client().prepareIndex(ANOTHER_INDEX).setId(Integer.toString(i)).setSource("field", "value" + i).execute().actionGet();
+        }
+
+        refresh(INDEX_NAME, ANOTHER_INDEX);
+        assertBusy(() -> {
+            assertHitCount(client(replicaNode).prepareSearch(INDEX_NAME).setSize(0).setPreference("_only_local").get(), initialDocCount);
+            assertHitCount(client(replicaNode).prepareSearch(ANOTHER_INDEX).setSize(0).setPreference("_only_local").get(), initialDocCount);
+        });
+
+        SegmentReplicationStatsResponse segmentReplicationStatsResponse = client().admin()
+            .indices()
+            .prepareSegmentReplicationStats(INDEX_NAME, ANOTHER_INDEX)
+            .execute()
+            .actionGet();
+
+        // Verify that Segment Replication happened on the replica shard.
+        assertFalse(segmentReplicationStatsResponse.getReplicationStats().get(ANOTHER_INDEX).get(0).getReplicaStats().isEmpty());
```

Comment:
Rather than verifying with stats API, can we invoke GetSettings `client().admin().indices().getSettings and validate the settings value returned?  And validate the value from getIndexSettings.isSegrepEnabled?

Also on that note I think we should validate the output of GET _settings with/without defaults to ensure the correct replicationType is returned.